ports/gilbert.js

```diff
@@ -0,0 +1,655 @@
+// SPDX-License-Identifier: BSD-2-Clause
+// Copyright (c) 2024 abetusk
+
+"use strict";
+
+
+var gilbert = {
+  "xy2d": gilbert_xy2d,
+  "d2xy": gilbert_d2xy,
+
+  "xyz2d": gilbert_xyz2d,
+  "d2xyz": gilbert_d2xyz,
+};
+
+function sgn(x) {
+  if (x<0) { return -1; }
+  if (x>0) { return  1; }
+  return 0;
+}
+
+function inbounds2(p, s, a, b) {
```

Comment:
Perhaps `in_bounds` would be more readable than `inbounds` (here and elsewhere)

Reply:
will do